src/gui/qgsmapcanvas.cpp

```diff
@@ -778,8 +796,26 @@ QgsRectangle QgsMapCanvas::imageRect( const QImage& img, const QgsMapSettings& m
 
 void QgsMapCanvas::mapUpdateTimeout()
 {
-  const QImage& img = mJob->renderedImage();
-  mMap->setContent( img, imageRect( img, mJob->mapSettings() ) );
+  QImage img = mJob->renderedImage();
+  magnifyImage( img );
+
+  mMap->setContent( img, imageRect( img, mSettings ) );
+}
+
+void QgsMapCanvas::magnifyImage( QImage &img )
+{
+  if ( mMagnificationLevel != 1.0 )
+  {
+    img = img.scaled( img.width() * mMagnificationLevel,
```

Comment:
I've got to say - I honestly don't see the point in just seeing a pixelated zoom. Most operating systems already have a tool for this if all you need is just to zoom in to pixels. Ditto with high res displays - just resizing up an image isn't giving any extra detail.

But if we do this properly and instead render at a magnifier level then this becomes a killer cartographic tool. That's why I'm against the pixelated zoom approach. A DPI based zoom would still satisfy the "I just need to see things bigger" use case but also satisfy the needs of precise cartographers.

Its also not a major rewrite, just a matter of tweaking the map settings. All the framework is already there - composer uses it for instance.